salesagility/SuiteCRM
# Fix OPCache install module copy action

This PR fixes an OPCache issue where copy actions during module installation don't update the OPCache.

## Description
In module manifests copy actions can be implemented which causes SuiteCRM to copy files during installation of the module.
When these copy actions overwrite an existing file, OPCache is not notified.

On systems which have OPCache configured with:
`opcache.validate_timestamps=0`
the cache is not updated, causing PHP to run the previously overwritten file.

## Motivation and Context
This PR follows up on the following PRs which address OPCache issues:
https://github.com/salesagility/SuiteCRM/pull/8514
https://github.com/salesagility/SuiteCRM/pull/9628

The objective of these PRs is to make sure that performance optimized PHP configurations with (php.ini)
`opcache.validate_timestamps=0`
are able to run SuiteCRM without any hickups.

This PR solves one (hopefully last) issue in which OPCache clearance is implemented in the method:
ModuleInstaller::copy_recursive_with_backup
Analog to:
https://github.com/salesagility/SuiteCRM/pull/8514/files#diff-d2a1334de52ae090d4843f695abbad88ba7b1b5b387dfb4cbee6a11b6e3bf41d

## How To Test This

1. Configure PHP to have
    - opcache enabled
    - `opcache.validate_timestamps=0` (php.ini)
2. Open SuiteCRM
3. Click "Admin" -> "About"
    => The About page is shown: OK
4. Open "Admin" -> "Module Loader"
5. Load and install the following module, this module overwrites the file modules/Home/About.php
    [test_opcache_install_copy_about_page.zip](https://github.com/salesagility/SuiteCRM/files/10333293/test_opcache_install_copy_about_page.zip)
6. Click "Admin" -> "About"

**Expected Result**
A modified About page is shown with on top the text "TEST MODULE INSTALL COPY ACTION OPCACHE"

**Actual Behaviour**
The About page is unchanged and displays the previously overwritten version

## Types of changes


### Final checklist
